pkg/logs/metrics/utilization_monitor.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/VividCortex/ewma"
+)
+
+// UtilizationMonitor is an interface for monitoring the utilization of a component.
+type UtilizationMonitor interface {
+	Start()
+	Stop()
+}
+
+// NoopUtilizationMonitor is a no-op implementation of UtilizationMonitor.
+type NoopUtilizationMonitor struct{}
+
+// Start does nothing.
+func (n *NoopUtilizationMonitor) Start() {}
+
+// Stop does nothing.
+func (n *NoopUtilizationMonitor) Stop() {}
+
+// TelemetryUtilizationMonitor is a UtilizationMonitor that reports utilization metrics as telemetry.
+// Utilization is calculated as the ratio of time spent in use to the total time.
+// Utilization can change rapidly over time based on the workload. So the monitor samples the utilization over a given interval.
+type TelemetryUtilizationMonitor struct {
+	inUse      time.Duration
+	idle       time.Duration
+	startIdle  time.Time
+	startInUse time.Time
+	avg        ewma.MovingAverage
+	name       string
+	instance   string
+	ticker     *time.Ticker
+}
+
+// NewTelemetryUtilizationMonitor creates a new TelemetryUtilizationMonitor.
+func NewTelemetryUtilizationMonitor(name, instance string, interval time.Duration) *TelemetryUtilizationMonitor {
+	return &TelemetryUtilizationMonitor{
+		startIdle:  time.Now(),
+		startInUse: time.Now(),
+		avg:        ewma.NewMovingAverage(),
```

Comment:
Not sure we need a dependency to do `old*(1-factor)+new*factor`, but if we use one: what is the factor here, how fast it reacts to changes? Is it going to be stable between library versions? Would it make sense to make it configurable?

Reply:
> Not sure we need a dependency
It was already in our codebase - so I reached for it. Happy to reimplement it if preferred. 

> what is the factor here
The docs explain it [here](https://pkg.go.dev/github.com/VividCortex/ewma?utm_source=godoc#pkg-constants): `NewMovingAverage` uses the library defaults (one min window). 

>  Is it going to be stable between library versions?
[It hasn't changed in 11 years](https://github.com/VividCortex/ewma/blame/master/ewma.go#L7) - so seems pretty likely. 

> Would it make sense to make it configurable?
[It is via the constructor](https://github.com/VividCortex/ewma/blob/v1.2.0/ewma.go#L40) however the default seems good enough for now. 

These metrics are for internal use only at the moment to help us drive optimizations so we may choose to change the ewma settings in the future.

Reply:
Closing this since I removed this dep